Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,243 @@
+        }
+    }
+
+    // example of vent functionality
+    public IEnumerator Vent(Vent original)
+    {
+        Info("I entered a vent!");
+        List<Vent> possibleVents = GetAvailableNearbyVents(original);
+        Vent current = possibleVents[UnityEngine.Random.RandomRangeInt(0, possibleVents.Count)];
+        yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+        string error;
+        if (!original.TryMoveToVent(current, out error))
+        {
+            Error($"Failed to move to vent {current.Id}, reason: {error}");
+        }
+        while (true)
+        {
+            // use a random time between vent moves to make it more realistic
+            yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+            bool playerFound = false;
+            foreach (KeyValuePair<PlayerControl, LastSeenPlayer> player in NeuroPlugin.Instance.Vision.GetPlayerLocations())
+            {
+                if (PlayerControl.LocalPlayer == player.Key) continue;
+                if (player.Key.Data.IsDead) continue;
+                if (player.Key.Data.RoleType == AmongUs.GameOptions.RoleTypes.Impostor) continue;
+
+                // if we see a player in our radius, try a different vent
+                if (player.Value.time > Time.timeSinceLevelLoad - (2 * Time.fixedDeltaTime))
+                {
+                    Info($"Spotted {player.Key.name}, trying a different exit vent...");
+                    Vent next;
+                    while (true)
+                    {
+                        possibleVents = GetAvailableNearbyVents(current);
+                        next = possibleVents[UnityEngine.Random.RandomRangeInt(0, possibleVents.Count)];
+                        if (current == next) continue;
+                        break;
+                    }
+                    if (!current.TryMoveToVent(next, out error))
+                    {
+                        Error($"Failed to move to vent {next.Id}, reason: {error}");
+                    }
+                    current = next;
+                    playerFound = true;
+                    break;
+                }
+            }
+            // if we dont see anyone, exit the vent we're currently in
+            // also reset the kill target out of the vent to prevent any crossmap targeting
+            if (!playerFound)
+            {
+                HudManager.Instance.ImpostorVentButton.DoClick();
+                killTarget = null;
+                yield break;
+            }
+        }
+    }
+
+    // since some vents can have a variable amount of neighbors, use this helper method to get available ones
+    private List<Vent> GetAvailableNearbyVents(Vent vent)
+    {
+        List<Vent> result = new List<Vent>();
+        if (vent.Left)
+            result.Add(vent.Left);
+        if (vent.Center)
+            result.Add(vent.Center);
+        if (vent.Right)
+            result.Add(vent.Right);
+        return result;
```

Comment:
Use LINQ for this.

```suggestion
        return vent.NearbyVents.Where(v => v).ToList();
```

You can keep it as a method or inline it, it's up to you.